bundles/org.openhab.binding.guntamatic/src/main/java/org/openhab/binding/guntamatic/internal/GuntamaticHandler.java

```diff
@@ -0,0 +1,426 @@
+        } else {
+            String req = "http://" + config.hostname + url;
+            if (!config.key.isEmpty()) {
+                req += "?key=" + config.key;
+            }
+            for (int i = 0; i < params.length; i++) {
+                req += "&" + params[i];
+            }
+
+            Request request = httpClient.newRequest(req);
+            request.method(HttpMethod.GET).timeout(5, TimeUnit.SECONDS).header(HttpHeader.ACCEPT_ENCODING, "gzip");
+
+            try {
+                ContentResponse contentResponse = request.send();
+                if (contentResponse.getStatus() == 200) {
+                    if (!this.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                        updateStatus(ThingStatus.ONLINE);
+                    }
+                    try {
+                        String response = new String(contentResponse.getContent(), Charset.forName(config.encoding));
+                        if (url == DAQEXTDESC_URL) {
+                            parseAndJsonInit(response);
+                        } else if (url == DAQDATA_URL) {
+                            parseAndUpdate(response);
+                        } else if (url == DAQDESC_URL) {
+                            parseAndInit(response);
+                        } else if (url == PARSET_URL) {
+                            // via return
+                        }
+                        return response;
+                    } catch (IllegalArgumentException e) {
+                        errorReason = String.format("IllegalArgumentException: %s", e.getMessage());
+                    }
+                } else {
+                    errorReason = String.format("Guntamatic request failed with %d: %s", contentResponse.getStatus(),
+                            contentResponse.getReason());
+                }
+            } catch (TimeoutException e) {
+                errorReason = "TimeoutException: Guntamatic was not reachable on your network";
+            } catch (ExecutionException e) {
+                errorReason = String.format("ExecutionException: %s", e.getMessage());
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                errorReason = String.format("InterruptedException: %s", e.getMessage());
+            }
+        }
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorReason);
+        return null;
+    }
+
+    private void pollGuntamatic() {
+        // logger.warn("pollGuntamatic: initalized: {}", initalized);
+        if (initalized == false) {
+            if (!config.key.isEmpty()) {
+                sendGetRequest(DAQEXTDESC_URL);
+            }
+            sendGetRequest(DAQDESC_URL);
+        } else {
+            sendGetRequest(DAQDATA_URL);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(GuntamaticConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
```

Comment:
Such a check is already implemented in ...
```
    private @Nullable String sendGetRequest(String url, String... params) {
        String errorReason = "";
        if (config == null) {
            errorReason = "Invalid Binding configuration";
        } else if (config.hostname.isBlank()) {
            errorReason = "Invalid hostname configuration";
```

Nevertheless I implemented your proposal as it doesn't make sense to start the polling in case of a blank configuration:
```
        if (config.hostname.isBlank()) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Invalid hostname configuration");
        }
```